snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Tracker.java

```diff
@@ -957,4 +964,27 @@ public void removeGlobalContext(String tag) {
             }
         }
     }
+
+    // gdpr context
+    private boolean gdprContext = false;
+    private String basisForProcessing;
+    private String documentId;
+    private String documentVersion;
+    private String documentDescription;
+
+    /**
+     * Enables gdpr context to be sent with every event
+     */
+    public void enableGDPRContext(BASIS basisForProcessing, String documentId, String documentVersion, String documentDescription) {
```

Comment:
This function seems like it should only set the global values if `gdprContext = false`.  Otherwise if you are mid sending events when it is on and switch it off you might get a thread race condition.

Also worth making this function thread safe with tracking calls to ensure you cannot have any race conditions between these tracker variables.

Atomic booleans could be useful here!

Reply:
> This function seems like it should only set the global values if gdprContext = false. 

It mimics how our js tracker offers gdpr context. Are you sure on this? 

Thanks for catching thread-safety. I'll make the method `synchronized`.

Reply:
Mimic the same behaviour then but thread safety is really important for Tracker functions!